src/cluster/src/start/k8.rs

```diff
@@ -560,94 +632,47 @@ impl ClusterInstallerBuilder {
 /// `ClusterInstaller::new()`
 ///
 /// ```no_run
-/// # use fluvio_cluster::ClusterInstaller;
-/// let installer = ClusterInstaller::new()
-///     .build()
-///     .expect("should initialize installer");
-///
-/// // Installing Fluvio is asynchronous, so you'll need an async runtime
-/// let result = fluvio_future::task::run_block_on(async {
-///     installer.install_fluvio().await
-/// });
+/// # use fluvio_cluster::{ClusterInstaller, ClusterConfig, ClusterError};
+/// # async fn example() -> Result<(), ClusterError> {
+/// let config = ClusterConfig::builder()
+///     .with_chart_version("0.7.0-alpha.1")
+///     .build()?;
+/// let installer = ClusterInstaller::from_config(config)?;
+/// let _status = installer.install_fluvio().await?;
+/// # Ok(())
+/// # }
 /// ```
 ///
 /// [Helm Charts]: https://helm.sh/
 /// [Minikube]: https://kubernetes.io/docs/tasks/tools/install-minikube/
 #[derive(Debug)]
 pub struct ClusterInstaller {
     /// Configuration options for this installation
-    config: ClusterInstallerBuilder,
+    config: ClusterConfig,
```

Comment:
Do we need another layer of abstraction?  I would prefer we don't make changes to how config works until there is need or better abstraction.

Reply:
The reason for needing this change is that previously, we had no config options that did not have a default value. Now, it should be _invalid_ to produce a `ClusterConfig` without first providing a `chart_version`. However, we do not want to put `chart_version: Option<String>` directly into our Config object because then our business logic would need  to check that it is present at every usage point.

Now, we distinguish between the Builder, which is where we collect all of our options, and the Config, which is where we are guaranteed to have all of the options we need. To get from Builder to Config you use `.build()`, which will fail if not all of the options are provided.